spec/spec_helper.rb

```diff
@@ -48,6 +77,29 @@ def build_event(message)
 
 RSpec.configure do |c|
   c.include GrokHelpers
+  c.include GrokHelpers::Shared
+  c.extend GrokHelpers::Shared
+end
+
+def describe_pattern(name, pattern_modes = [ nil ], &block)
+  pattern_modes.each do |mode|
+    RSpec.describe "#{name}#{mode ? " (#{mode})" : nil}" do
+
+      before(:all) do
```

Comment:
🤔 I'm not the biggest fan of `before(:all)`, even though it can be a bit more efficient, because it requires ordered non-interleaved execution. Yeah, rspec executes context groups together and doesn't run specs in parallel unless some other testing harness is hooked into, but this "feels" like a sharp edge to me.

Reply:
okay thanks, refactored - no preference really on my end.